### PR TITLE
Fix SampleLog units for MergeRun, ConjoinXRuns

### LINKED ISSUE #22700: MergeRuns missing units in results SampleLogs
You need a script that merges two entities having units like this:
```
ws1 = LoadMLZ(Filename='TOFTOFTestdata.nxs')
ws2 = LoadMLZ(Filename='TOFTOFTestdata.nxs')

# change sample logs for a second workspace, not needed for real workspaces
AddSampleLog(ws1, 'temperature', str(294.14), 'Number')
lognames = 'temperature,run_start,run_end,monitor_counts,run_number'
logvalues = '296.16,2013-07-28T11:32:19+0053,2013-07-28T12:32:19+0053,145145,TOFTOFTestdata2'
AddSampleLogMultiple(ws2, lognames, logvalues)

group=GroupWorkspaces('ws1,ws2')
m2=MergeRuns(group)
```

If you inspect the `SampleLogs`, units are missing for entries duration and temperature.

See also https://github.com/mantidproject/mantid/pull/22677

### PR DESCRIPTION
**Pull request by vreimund**
**Description of work.**

https://github.com/mantidproject/mantid/pull/23965 needs to be merged first.

`MergeRuns`, `ConjoinXRuns` did not propagate the sample log units. Now, units are present for sample log entries (sum, list, time series merging).

Somhow related issue:
https://github.com/mantidproject/mantid/issues/24006

**To test:**
Please see the issue  for test examples

Fixes #22700.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by DTasev** [comment]
I'll review this after #23965 has been merged

**Comment by DTasev** [comment]
I've unassigned myself as I can't immediately start looking at this - if someone wants to pick it up go ahead. I'll come back to it before the end of this week

File: docs/source/release/v3.14.0/framework.rst [hunk 1]
```diff
@@ -84,6 +84,7 @@ Bugfixes
 - :ref:`Fit <algm-Fit>` now applies the ties in correct order independently on the order they are set. If any circular dependencies are found Fit will give an error.
 - Fixed a rare bug in :ref:`MaskDetectors <algm-MaskDetectors>` where a workspace could become invalidated in Python if it was a ``MaskWorkspace``.
 - Fixed a crash in :ref:`MaskDetectors <algm-MaskDetectors>` when a non-existent component was given in ``ComponentList``.
+- SampleLogs units are present, algorithms :ref:`MergeRuns <algm-MergeRuns>` and :ref:`ConjoinXRuns <algm-ConjoinXRuns>`.
```
Comment (DTasev) [inline docs/source/release/v3.14.0/framework.rst]: Can this be made more clear? Something like "the output workspace now keeps the units of the input workspace"

**Review by dtasev — CHANGES_REQUESTED** [review]
Minor comment about clarifying the release notes. Apart from that everything looks good

File: docs/source/release/v3.14.0/framework.rst (hunk 1 above)
Comment (reimundILL) [reply docs/source/release/v3.14.0/framework.rst]: Sure.

**Review by dtasev — APPROVED** [review]

**Pull request merged**